src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case "addProperties":
+        currentState = Object.assign({}, currentState, action.extraData);
+        result.push(Object.assign({}, currentState));
+        break;
+      case "removeProperties":
+        for (const key of action.keysToRemove) {
+          if (currentState.hasOwnProperty(key)) {
+            delete currentState[key];
+          }
+        }
+        result.push(Object.assign({}, currentState));
+        break;
+      case "clear":
+        currentState = {};
+        result.push(Object.assign({}, currentState));
```

Comment:
[checklist ](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md)point 4, we can also just spread state clone 😄